### DISCUSSION TITLE
How to build openSSL for arm64 device and arm64 simulator in M1 mac ?

### DISCUSSION THREAD

**Discussion by VaraVLDev** [openssl/openssl#25688] [Q&A]
Command used for generating library for arm64 simulator architecture

`./Configure iossimulator-xcrun "-arch arm64" no-tests no-asm no-shared no-module no-hw no-async enable-fips`

Command used for generating library for arm64 device architecture

`./Configure ios64-xcrun no-tests no-shared no-module no-asm enable-fips`

**Unable to create fat library as both having same architecture arm64**

`% lipo -create libssl_arm64_Device.a libssl_arm64_Simulator.a -output libssl.a`

**fatal error:** /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/lipo: libssl_arm64_Device.a and libssl_arm64_Simulator.a **have the same architectures (arm64) and can't be in the same fat output file**

**Goal:**
Trying to generate OpenSSL library for an iOS application which can run on arm64 simulator and arm64 device.

**Observations:**
Even though architectures are same, building for device will not work in arm64 simulator and vice versa.

Please suggest if any other way possible to build openSSL which can work on both device and simulator.

**Comment by tom-cosgrove-arm** [comment]
What are you trying to achieve? If you want to create a fat binary containing OpenSSL for both arm64 device and arm64 simulator you can't - fat binaries are to have the same builds but for different architectures, not different builds for the same architectures.

As to how to do that... that's a question about iOS development, not OpenSSL.

**Reply by VaraVLDev** [reply]
Goal:
Trying to generate OpenSSL library for an iOS application which can run on arm64 simulator and arm64 device.

Observations:
Even though architectures are same, building for device will not work in arm64 simulator and vice versa.

Please suggest if any other way possible to build openSSL which can work on both device and simulator.

**Comment by addsoa** [comment]
I have the similar issue or question. I want to cross-compile openssl for mac-arm. The building passed when I built by 
``` bash
./Configure darwin64-arm64 $Other-cross-compile-configures
```
While I've noticed that in `Configurations/10-main.conf` 
``` txt
    "darwin64-x86_64-cc" => { inherit_from => [ "darwin64-x86_64" ] }, # Historic alias
    "darwin64-x86_64" => {
        inherit_from     => [ "darwin-common" ],
        CFLAGS           => add("-Wall"),
        cflags           => add("-arch x86_64"),
        lib_cppflags     => add("-DL_ENDIAN"),
        bn_ops           => "SIXTY_FOUR_BIT_LONG",
        asm_arch         => 'x86_64',
        perlasm_scheme   => "macosx",
    },
    "darwin64-arm64-cc" => { inherit_from => [ "darwin64-arm64" ] }, # "Historic" alias
    "darwin64-arm64" => {
        inherit_from     => [ "darwin-common" ],
        CFLAGS           => add("-Wall"),
        cflags           => add("-arch arm64"),
        lib_cppflags     => add("-DL_ENDIAN"),
        bn_ops           => "SIXTY_FOUR_BIT_LONG",
        asm_arch         => 'aarch64',
        perlasm_scheme   => "ios64",
    },
```
The `perlasm_scheme` is `ios64` instead of `macosx` as that of `darwin64-x86_64`.  

Is that the way it should be or is that the compatible solution when native arm64 application was not mainstream for macos apple-chipset?
